hack/generator/pkg/functions/pivot_conversion_function.go

```diff
@@ -0,0 +1,204 @@
+
+	"github.com/Azure/azure-service-operator/hack/generator/pkg/astbuilder"
+	"github.com/Azure/azure-service-operator/hack/generator/pkg/astmodel"
+	"github.com/Azure/azure-service-operator/hack/generator/pkg/conversions"
+)
+
+// PivotConversionFunction implements a pivot that's used when a conversion reaches a hub type but hasn't yet found the
+// innermost required conversion needed.
+//
+// We use this for Spec types, implementing genruntime.ConvertibleSpec, and for Status types, implementing
+// genruntime.ConvertibleStatus.
+//
+// If the two instances involved in conversion are not in the same "spoke" leading to this "hub" version, we'll pivot to
+// the reverse conversion, starting at the far end of that "spoke":
+//
+// func (s <hubtype>) ConvertFrom(instance <interfaceType>>) error {
+//     return instance.ConvertTo(s)
+// }
+//
+// The pivot is needed when following the package references from resource version to resource version won't lead us to
+// encounter the other type that's involved in the conversion, as we can see if we're trying to convert between v1 and
+// v2:
+//
+//      ##################                  #################                  +---------------+
+//      #       v1       #                  #       v2      #                  |       v3      |
+//      #     Person     #                  #     Person    #                  |     Person    |
+//      ##################                  #################                  +---------------+
+//              |                                   |                                  |
+//              v                                   v                                  v
+//      +----------------+                  +---------------+                  +---------------+
+//      |    v1storage   |                  |    v2storage  |                  |    v3storage  |
+//      |     Person     |----------------->|     Person    |----------------->|     Person    |
+//      +----------------+                  +---------------+                  +---------------+
+//
+// Following package references from v1 leads us, in turn, to v1storage, v2storage, and finally v3storage (our hub
+// version), none of is the version we need to terminate the conversion path.
+//
+// By having a pivot conversion on v3, we can restart the conversion path with v2.
+//
+// The entire conversion works as follows:
+//
+// v1.Person.ConvertTo(v2.Person)
+// --> v1storage.Person.ConvertTo(v2.Person)
+//   --> v2storage.Person.ConvertTo(v2.Person)
+//     --> v3storage.Person.ConvertTo(v2.Person)              // Pivot!
+//       --> v2.Person.ConvertFrom(v3storage.Person)          // Change of direction
+//         --> v2storage.Person.ConvertFrom(v3storage.Person)
+//
+// Note that conversions like this always pivot through the hub version.
+//
+type PivotConversionFunction struct {
+	// nameFrom is the name for this function when converting FROM a provided instance
+	nameFrom string
```

Comment:
I wrote this up before you refactored but then I think it was lost when this file got renamed, but you could instead of having a `nameFrom`/`nameTo`, just assign a single `name` field in the `PivotConversionFunction` constructors.

Something like:
`name: direction.SelectString("ConvertSpecFrom", "ConvertSpecTo")`

It feels a bit weird to have the name dynamically selected at generation time when it's actually known significantly earlier (at constructor called time).

Minor though.

Reply:
Oh wait apparently that comment was saved - I see it above. It applies here too though I guess.
Github weird after refreshing in the middle of putting comments I guess.

Reply:
No longer need both names, so will simplify as suggested.